src/Controllers/RegisterController.php

```diff
@@ -92,8 +92,10 @@ public function registerAction(): RedirectResponse
         }
 
         // Set the user active
-        $user->active = true;
-        $users->save($user);
+        auth()->activateUser($user);
+
+        // a successful login
+        Events::trigger('login', $user);
```

Comment:
I don't think `login` event should be fired during registration.

Reply:
Registration without email activation logs in the user (call `login()` method).
The user can do without any restriction.

If we think registration is not login, and should not fire a login event,
then what do we do after email activation?

Reply:
I think that line can be deleted, actually. We have a `didRegister` event fired above (which should probably be renamed to simply `register` to match the login ones. And the `login()` method of the Session handler now fires off a login event for us. This is now redundant, I believe.

Reply:
I removed firing login event in `Session::login()`, because when we use 2FA login is not completed at that point.

Reply:
Oh, I missed that point. The latest code I pulled down yesterday had it in there I'm pretty sure.

https://github.com/codeigniter4/shield/blob/develop/src/Authentication/Authenticators/Session.php#L290